lib/io.js

```diff
@@ -61,14 +56,7 @@ BinaryDecoder.prototype = {
         var numShifts = 0;
 
         while (b.greaterThan(L_0x7f) && numShifts < 7) {
-            b = this.readByte();
-            if (b instanceof AvroErrors.BlockDelayReadError) {
-                //console.error("offset now is %d, old was %d, shift is %d", this._input.offset, oldOffset, shift);
-                this._input.skip(oldOffset - this._input.offset);
```

Comment:
This was weird - so, if `readByte()` returns an error, it moves the offset to skip that block from the Buffer? Sounds dangerous. From what I remember, removing this was due to a change you made in which it throws the error, rather than returning it, correct?